File: Python/Non-Constructible Change/solution_1.py
coins = [5,7,1,1,2,3,22]

def nonConstructibleChange(coins):

    increase = 1
    created_coins = set()
    

    for i in range(len(coins)):
        for j in range(1,len(coins)):

            created = coins[i] + coins[j]
            created_coins.add(created)

    
    lst = sorted(created_coins)

    if len(lst) == 0:
        return 1

    for element in lst:
        if element + increase not in lst:
            return element + increase

print(nonConstructibleChange(coins))

File: Go/Non-Constructible Change/solution_1.go
package main

import "sort"

func NonConstructibleChange(coins []int) int {

	sort.Ints(coins)

	currentChangeCreated := 0

	for _, coin := range coins {

		if coin > currentChangeCreated+1 {
			return currentChangeCreated + 1
		}

		currentChangeCreated += coin

	}

	return currentChangeCreated + 1
}
